src/RelayProvider.ts

```diff
@@ -0,0 +1,278 @@
+         * In order to avoid this, we created a new field called forceGas, which is optional.
+         * If the user wants to manually set the gas, then she can put a value in the forceGas field, otherwise it will be
+         * automatically estimated (correctly) by the RelayClient.
+         * The reason why we added this new forceGas field is because at this point, by reading the gas field,
+         * we cannot differentiate between a user-entered gas value from the auto-calculated (and a very poor calculation)
+         * value that comes from the original provider
+         */
+
+        const fullEnvelopingRequest: UserDefinedEnvelopingRequest = {
+            relayData: {
+                ...envelopingRequest.relayData,
+                callForwarder: callForwarderValue,
+            }, 
+            request: {
+                ...envelopingRequest.request,
+                relayHub: relayHubValue,
+                gas: gasToSend,
+            }
+        }
+
+        const fullRequestConfig: RequestConfig = {
+            ...requestConfig,
+            onlyPreferredRelays: onlyPreferredRelaysValue
+        }
+
+        const relayingResult = await this.executeRelayTransaction(fullEnvelopingRequest, fullRequestConfig);
+
+        if(relayingResult.receipt){
+            const { relayRevertedOnRecipient, reason } = this._getRelayStatus(relayingResult.receipt);
+            const { transactionHash } = relayingResult.receipt;
+
+            if (relayRevertedOnRecipient) {
+                throw new Error(
+                        `Transaction Relayed but reverted on recipient - TxHash: ${transactionHash} , Reason: ${reason ?? 'unknown'}`
+                );
+            }
+        }
+
+        return relayingResult;
+    }
+
+    override async listAccounts() {
+        let accountsList = await this.jsonRpcProvider.listAccounts();
+
+        if (accountsList && Array.isArray(accountsList)) {
+            const { chainId } = await this.getNetwork();
+            const accountManager = new AccountManager(this, chainId);
+
+            const ephemeralAccounts =
+                accountManager.getAccounts();
+
+            accountsList = accountsList.concat(ephemeralAccounts);
+        }
+
+        return accountsList;
+    }
+
+    override send(method: string, params: Array<Record<string, unknown>>): Promise<unknown> {
+        if (this._useEnveloping(method, params)) {
+            const { requestConfig, envelopingTx } = params[0] as { requestConfig: RequestConfig, envelopingTx: UserDefinedEnvelopingRequest }
+            const { request } = envelopingTx;
+            if (method === 'eth_sendTransaction') {
+
+                if (!request.to) {
+                    throw new Error(
+                        'Relay Provider cannot relay contract deployment transactions. Add {from: accountWithRBTC, useEnveloping: false}.'
```

Comment:
Im not sure about this, on smart wallet deployments, the to can be undefined.

Reply:
From what I understood, the send method of the relayProvider should not be used to deploy smart wallets.

Reply:
Yes, indeed. See the [previous code](https://github.com/rsksmart/rif-relay-client/blob/main/src/RelayProvider.ts#L125).